docs/coding/csharp.md

```diff
@@ -1,141 +1,363 @@
 # C# Coding Standards
 
+## Sections
+- [Naming](#naming)
+  - [Abbreviations](#abbreviations)
+  - [Casing](#casing)
+  - [Namespaces](#namespaces)
+  - [Types](#types)
+  - [Methods](#methods)
+  - [Variables and Fields](#variables-and-fields)
+  - [Miscellaneous](#miscellaneous)
+
+- [Layout](#layout)
+  - [Comments](#comments)
+  - [Files](#files)
+  - [Whitespace](#whitespace)
+
+- [Coding](#coding)
+  - [Arrays](#arrays)
+  - [Assemblies](#assemblies)
+  - [Branches](#branches)
+  - [Classes](#classes)
+  - [Conversions](#conversions)
+  - [Dipsose()](#dipsose)
+  - [Enums](#enums)
+  - [Equality](#equality)
+  - [Exceptions](#exceptions)
+  - [Fields](#fields)
+  - [Flags](#flags)
+  - [Increment/decrement](#incrementdecrement)
+  - [Interfaces](#interfaces)
+  - [Methods](#methods-1)
+  - [Parameters](#parameters)
+  - [Properties](#properties)
+  - [Strings](#strings)
+  - [Structs](#structs)
+  - [Synchronization](#synchronization)
+  - [Tasks](#tasks)
+  - [Threads](#threads)
+  - [ToString()](#tostring)
+  - [Types](#types)
+  - [Miscellaneous](#miscellaneous)
+
+
+
 ## Naming
 
+### Abbreviations
+- :x: DO NOT: use abbreviations or contractions within identifier names.
+
 ### Casing
-- :heavy_check_mark: DO: Use ‘camelCasing’ for variables and method arguments
-- :heavy_check_mark: DO: Use ‘PascalCasing’ for everything else including namespaces, interfaces, types, methods, properties, events, etc.
+- :heavy_check_mark: DO: capitalize both characters in two-character acronyms, except for the first word of a camelCased identifier.
+- :heavy_check_mark: DO: capitalize only the first character in acronyms with three or more characters, except for the first word of a camelCased identifier.
+- :x: DO NOT: capitalize any of the characters in acronyms at the beginning of a camelCased identifier.
+- :heavy_check_mark: DO: use PascalCasing for all class names.
+- :heavy_check_mark: DO: use camelCasing for local variable names.
+- :heavy_check_mark: DO: use uppercase literal suffixes (e.g., 1.618033988749895M).
+- :heavy_check_mark: DO: use camelCasing for variable declarations using tuple syntax. 
+- :heavy_check_mark: CONSIDER: using PascalCasing for all tuple item names.
+- :heavy_check_mark: DO: use PascalCasing for namespace names.
+- :heavy_check_mark: DO: use camelCasing for parameter names.
+- :heavy_check_mark: CONSIDER: using the same casing on a propertyâ€™s backing field as that used in the property, distinguishing the backing field with an â€œ_â€ prefix. Do not, however, use two underscores; identifiers beginning with two underscores are reserved for the use of t.
```

Comment:
Can we filter out the "smart" quotes that are coming from the CSV file? `with an â€œ_â€ prefix`